### DISCUSSION TITLE
Problem with rename notes

### DISCUSSION THREAD

**Discussion by Cryptodarktrader** [SilentVoid13/Templater#1221] [Help] [answered]
I'm trying to create a template for cooking recipes.

The idea is that when inserting the template it asks you for the name of the recipe, and notes it as a property:

```
---
title: <% await tp.system.prompt("Recipe Name?") %>
---
```

My problem comes when I try to modify the title of the note to the one I just included in YAML and then move that note to the corresponding folder using:

<% await tp.file.move("/20 - 🍜 Recipes/" + await tp.frontmatter["title"]) %>

The note moves me but it doesn't change the title. Does anyone know how I can do it?

**Comment by Zachatoo** [comment, marked as answer]
I'd recommend storing your new title in a variable and accessing it from that instead. `tp.frontmatter` gets populated at time of execution start, so `tp.frontmatter["title"]` won't be available if you change the frontmatter while you're executing the template.

```js
<%*
const title = await tp.system.prompt("Recipe Name?");
await tp.file.move("/20 - 🍜 Recipes/" + title);
-%>
---
title: <% title %>
---
```

**Comment by Cryptodarktrader** [comment]
wow, that worked perfectly and it has also helped me correct similar errors in other types of files, I thank you very much!!
